### PR TITLE
[JENKINS-70599] restore installNecessaryPlugins redirect destination

### PR DESCRIPTION
**Pull request by MarkEWaite**
## [JENKINS-70599](https://issues.jenkins.io/browse/JENKINS-70599) restore installNecessaryPlugins redirect destination

The installNecessaryPlugins API end point previously redirected to the "/updateCenter" URL.  When the side bar was added to the plugin manager, the redirect was changed to "/updates".  Unfortunately, there is no page at "/updates", while there is a page at "/updateCenter".

This restores the redirect for the installNecessaryPlugins API call.

If the redirect is not restored, calls to the API that follow the redirect fail with a page not found (HTTP 404).

I did not find any callers to that API from inside Jenkins core.

Change introduced in 6db88c3f5ba82ecc68af2bced3a568058cf80e2b

See [JENKINS-70599](https://issues.jenkins.io/browse/JENKINS-70599).

### Testing done

* Interactively tested with `curl --location` to confirm that without this change, the call to `installNecessaryPlugins` redirects to a non-existent page.  Testing details are in the [Jira issue](https://issues.jenkins.io/browse/JENKINS-70599)
* Interactively tested with `curl --location` to confirm that with this change, the call to `installNecessaryPlugins` redirects to a valid page.  Testing details are in the [Jira issue](https://issues.jenkins.io/browse/JENKINS-70599)
* Automated test added to check the redirect URL is found after the fix and is not found before the fix.  Automated test uses the same technique as was used in the interactive test (request a crumb and a cookie, submit POST request with crumb and cookie)

### Proposed changelog entries

- Restore `installNecessaryPlugins` redirect destination.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@janfaracik or @NotMyFault 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7653"></a>

### CODE REVIEW

File: core/src/main/java/hudson/PluginManager.java
Comment (MarkEWaite) [inline core/src/main/java/hudson/PluginManager.java:2228]: This could be changed to `pluginManager/updates/` if it is important to include the `pluginManager` portion in the path of the redirect.  Both `/updateCenter` and `/pluginManager/updates/` exist.
```suggestion
        return HttpResponses.redirectViaContextPath("updateCenter");
```

**Comment by MarkEWaite** [comment]
@janfaracik I'd very much like to include this in the weekly release that will be delivered tomorrow so that it can be considered for backporting into Jenkins 2.387.1.  The change in the production code uses the same URL as was used before the regression.  The test is checking a method that test automation had never called before.

**Comment by janfaracik** [comment]
> @janfaracik I'd very much like to include this in the weekly release that will be delivered tomorrow so that it can be considered for backporting into Jenkins 2.387.1. The change in the production code uses the same URL as was used before the regression. The test is checking a method that test automation had never called before.
> 
> As far as I can tell, the method that is being changed is only used through the REST API and is never called from any Jelly page.

Hey, to be honest I know very little about this area of Jenkins being changed. I believe it was either @daniel-beck or @timja that had made those changes originally in #6783.

**Comment by MarkEWaite** [comment]
> Hey, to be honest I know very little about this area of Jenkins being changed. I believe it was either @daniel-beck or @timja that had made those changes originally in #6783.

Thanks!  I've switched the reviewers to @timja and @daniel-beck .  I feel quite confident in the change, since it has been confirmed by automated tests and by one of the two users that reported the issue.

File: core/src/main/java/hudson/PluginManager.java
Comment (timja) [reply core/src/main/java/hudson/PluginManager.java:2228]: this url is the more appropriate url to use (pluginManager/updates/)

As the updateCenter page shouldn't be linked / used anymore as far as I know

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Looks reasonable, thanks for taking care of.

/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
